File: FrogRiverOne/frog_river_one.cpp
#include "frog_river_one.h"

#include <set>
#include <vector>

int frog_river_one::solution(const int X, std::vector<int>& A)
{
    std::set<int> s;
    for (auto i = 0; i < A.size(); i++)
    {
        s.insert(A[i]);
        if (s.size() == X) return i;
    }

    return -1;
}

File: csharp/FrogRiverOne/FrogRiverOne.cs
using System.Collections.Generic;

public static class FrogRiverOne
{
    public static int Solution(int X, int[] A)
    {
        var sorted = new HashSet<int>(); // It is possible to use SortedSet<> and avoid line 11,12
        
        for (int i = 0; i < A.Length; i++)
        {         
            sorted.Add(A[i]);
            if (sorted.Count == X) return i;
        }

        return -1;
    }
}